src/Tribe/Tickets_View.php

```diff
@@ -1080,6 +1080,13 @@ public function get_rsvp_block( $post, $echo = true ) {
 		/** @var Tribe__Tickets__RSVP $rsvp */
 		$rsvp = tribe( 'tickets.rsvp' );
 
+		// Check if the call is coming from a shortcode.
+		$doing_shortcode = tribe_is_truthy( array_search( 'do_shortcode', array_column( debug_backtrace(), 'function' ) ) );
+
+		// Get the RSVP block HTML ID.
+		$block_html_id  = 'rsvp-now';
+		$block_html_id .= $doing_shortcode ? '-' . uniqid() : '';
```

Comment:
Could this just use the `$post_id`? Is there any possible way there would be multiple RSVP forms listed for the same post?

Reply:
We would also just need to adjust our rsvp-now links to use the post ID it's for.

Reply:
> Could this just use the $post_id? Is there any possible way there would be multiple RSVP forms listed for the same post? 
If we use the post ID, for the cases where they use the shortcode on an event (or CPT, where they created the RSVP) it'll be the same as having `rsvp-now`, because it'll be fixed. 

If they have a sales page with a builder, and then they have the RSVP block in different places, then they'll also have the duplicate ID if we rely on the `$post_id`.

So there are several cases where using `$post_id` would be the same as having a fixed string. I used `uniqueid()` so if it's coming from a shortcode, it always have a unique ID and that way we make sure that we don't have a duplicate ID. And have the fixed string `rsvp-now` for the cases in which we're including the block into the enabled CPTs (event, page, post, etc).

Reply:
I was just thinking over whether we could find a solution outside of needing to find out if we're doing the shortcode. I think we can move forward as-is if Luca thinks the `debug_backtrace()` is okay.

Reply:
✅